src/main/kotlin/brs/db/sql/SqlVersionedBatchEntityTable.kt

```diff
@@ -80,26 +81,25 @@ internal abstract class SqlVersionedBatchEntityTable<T> internal constructor(
         }
 
         dp.db.useDslContext { ctx ->
-            val updateQuery = ctx.updateQuery(table)
-            updateQuery.addValue(latestField, false)
-            for (idColumn in dbKeyFactory.pkColumns) {
-                updateQuery.addConditions(table.field(idColumn, Long::class.java).eq(0L))
-            }
-            updateQuery.addConditions(latestField?.isTrue)
-
-            // TODO can we avoid batching the single query here?
-            val updateBatch = ctx.batch(updateQuery)
-            for (dbKey in keySet) {
-                val pkValues = dbKey.pkValues
-                val bindArgs = arrayOfNulls<Any>(pkValues.size + 1)
-                bindArgs[0] = false
-                for (i in pkValues.indices) {
-                    bindArgs[i + 1] = pkValues[i]
+            // keySet chunked due:
+            // [SQLITE_ERROR] SQL error or missing database (Expression tree is too large (maximum depth 1000)
+            for (keySetChunk in keySet.chunked(990)) {
+                val updateQuery = ctx.updateQuery(table)
+                updateQuery.addConditions(latestField?.isTrue)
+                updateQuery.addValue(latestField, false)
+                var accountsCondition = DSL.noCondition()
+                for (dbKey in keySetChunk) {
+                    var pkCondition = DSL.noCondition()
+                    for ((index, idColumn) in dbKeyFactory.pkColumns.withIndex()) {
+                        pkCondition = pkCondition
+                            .and(table.field(idColumn, Long::class.java).eq(dbKey.pkValues[index]))
+                    }
+                    accountsCondition = accountsCondition.or(pkCondition)
                 }
-                updateBatch.bind(bindArgs)
+                updateQuery.addConditions(accountsCondition)
```

Comment:
UPDATE account SET latest=0 WHERE latest=1 AND ( (firstPK = <> AND secondPK = <> ...) OR (firstPK = <> AND secondPK = <> ...) ... )